hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -532,54 +533,67 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, nil
 	}
 
-	md := machineDeployment(nodePool, infraID, controlPlaneNamespace)
-	if result, err := controllerutil.CreateOrPatch(ctx, r.Client, md, func() error {
-		return r.reconcileMachineDeployment(
-			log,
-			md, nodePool,
-			userDataSecret,
-			machineTemplate,
-			infraID,
-			targetVersion, targetConfigHash, targetConfigVersionHash)
-	}); err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to reconcile MachineDeployment %q: %w",
-			client.ObjectKeyFromObject(md).String(), err)
+	// user provided infrastructure should not have any machine level cluster-api components (user provides infrastructure)
+	if nodePool.Spec.Platform.Type == hyperv1.IBMCloudPlatform && nodePool.Spec.Platform.IBMCloud != nil && nodePool.Spec.Platform.IBMCloud.ProviderType == configv1.IBMCloudProviderTypeUPI {
```

Comment:
Let's keep the scope of this change as small as possible just to enable you to make progress and let us having a design discussion and agreement before introducing bigger structural changes.

Let's model this from a behavioural pov:

```
func isAutomatedMachineManagement() bool {
	return !isIBMUPI()
}

func isIBMUPI() bool {
	return nodePool.Spec.Platform.IBMCloud != nil && nodePool.Spec.Platform.IBMCloud.ProviderType == configv1.IBMCloudProviderTypeUPI
}
```

Now let's just return early instead of if/else.

```
// TODO: design path forward for this scenario e.g IBM satellite
// consider eventual convergence with agent based workflow.
if !isAutomatedMachineManagement {
	return ctrl.Result{}, nil
}
```

Reply:
done